src/modals/cluster-upgrade/SelectHosts.js

```diff
@@ -1,215 +1,100 @@
-
-  onSelectRow (event, rowData) {
-    const selectedHostIdsSorted = this.state.sortedHosts
-      .filter(host => host.id === rowData.id ? !rowData.selected : this.state.selectedHostIds[host.id])
-      .map(host => host.id)
-
-    this.props.onChange(selectedHostIdsSorted)
-  }
-
-  onSelectAllRows (event) {
-    const newSelected = event.target.checked
-    this.props.onChange(newSelected ? this.state.sortedHosts.map(host => host.id) : [])
-  }
-
-  render () {
-    const { hosts } = this.props
-
-    if (!hosts || hosts.length === 0) {
-      return (
-        <div className='clusterUpgradeWizard-SelectHosts-NoHosts'>
-          <h2>{this.props.noHostsMessage}</h2>
-        </div>
-      )
-    }
-
+const SelectHosts = ({
+  hosts = [],
+  selectedHostIds = [],
+  onChange, // note: the function takes the complete set of selected host ids every time
+}) => {
+  if (!hosts || hosts.length === 0) {
     return (
-      <div className='clusterUpgradeWizard-SelectHosts'>
-        <Alert type='info' style={{ margin: '0' }}>
-          {this.props.selectHostsMessage}
-        </Alert>
-
-        <div className='tableContainer'>
-          <Table.PfProvider
-            striped
-            bordered
-            hover
-            columns={this.columns}
-          >
-            <Table.Header />
-            <Table.Body
-              rowKey='id'
-              rows={hosts.map(host => ({
-                id: host.id,
-                selected: this.state.selectedHostIds[host.id],
-                status: host.status,
-                name: host.name,
-                address: host.address,
-                vmInfo: host.summary,
-              }))}
-            />
-          </Table.PfProvider>
-        </div>
+      <div className='clusterUpgradeWizard-SelectHosts-NoHosts'>
+        <Title headingLevel='h2'>{msg.clusterUpgradeNoHostsMessage()}</Title>
       </div>
     )
   }
-}
 
-SelectHosts.i18nProps = {
-  noHostsMessage: 'There are no hosts in this cluster.  Can\'t upgrade nothing.',
+  const onSelectRow = (host, event, isSelected) => {
+    onChange(isSelected
+      ? [host.id, ...selectedHostIds]
+      : selectedHostIds.filter(id => id !== host.id)
+    )
+  }
 
-  selectHostsMessage:
-  // 'The order of the Hosts listed below will by the order that they' +
-  // 'are upgraded. Use sorting to affect the upgrade order.',
-    'If a host has a down status, it can cause the cluster upgrade to fail.',
+  const onSelectAll = (event, isSelected) => {
+    onChange(isSelected ? hosts.map(host => host.id) : [])
+  }
 
-  hostTableHeaderStatus: 'Status',
-  hostTableHeaderName: 'Name',
-  hostTableHeaderHostname: 'Hostname/IP Address',
-  hostTableHeaderVMs: 'Virtual Machines',
+  const allRowsSelected = hosts.every(host => selectedHostIds.includes(host.id))
+  const isThisHostSelected = host => allRowsSelected || selectedHostIds.includes(host.id)
+  const hostSortCompare = (host1, host2) => host1.name.localeCompare(host2.name, currentLocale())
```

Comment:
just in case  host or name is undefined please use `?.`

Reply:
That won't help anything.  If those fields are missing, the entire wizard will not work.